dashboard/app/models/course_offering.rb

```diff
@@ -457,4 +467,29 @@ def high_school_level?
 
     true
   end
+
+  def get_available_resources(locale_code='en-us')
+    lessons = latest_published_version(locale_code).units.first.lessons
+
+    return nil if lessons.empty?
+    lesson_plan = lessons.first.lesson_plan_html_url
+    expanded_card_resources = {"Lesson Plan" => lesson_plan}
+
+    lessons.each do |lesson|
+      break if expanded_card_resources.size >= 5
+      resources = lesson.resources
+
+      resources.each do |resource|
+        properties = resource.properties
+        next unless properties.key?('type')
+        type = properties['type']
+        type = "Slide Deck" if type == "Slides"
+        type = "Answer Key" if type == "Exemplar"
+        if ACCEPTABLE_RESOURCE_TYPES.include?(type) && expanded_card_resources.exclude?(type)
```

Comment:
Can you confirm that `expanded_card_resources.exclude?(type)` is properly excluding a type? I'm wondering if it would always return `true` since I imagine `Object.exclude?` is expecting a key:value pair rather than a single key. So you may need to check if the object's keys contains the given key rather than the object itself?

Reply:
Looks like this may be the case, ill use the has_key? method instead!